data/users.js

```diff
@@ -130,28 +133,32 @@ const updateUser = async (userId, fName, lName, username, password) => {
 	*/
   username = validation.checkUsername(username);
   password = validation.checkPassword(password, "Password");
+  const passwordHash = await bcrypt.hash(password, saltRounds);
 
   const existingUser = await getUserById(userId);
 
   const updatedUserInfo = {
     fName: fName,
     lName: lName,
     username: username,
-    password: password,
+    password: passwordHash,
   };
 
   // Checks if any of the fields are different from what already exists for user in DB
   checkUpdate(existingUser, updatedUserInfo);
   // Checks if the username that the user wants to change to is already in DB (used by another user)
   // Try to see if the username already exists in DB
-  let existingUsername = undefined;
-  try {
-    existingUsername = await getUserByUsername(username);
-  } catch (error) {
-    // Catch the error thrown if username is not found in db
-    // But don't do anything since, it means the username is available
+  if (existingUser.username !== updatedUserInfo.username) {
```

Comment:
This is checking if the existing user's username is the same as the updated information, but this is handled already in checkUpdate function,